File: 0228-summary-ranges/0228-summary-ranges.cs
public class Solution {
    public IList<string> SummaryRanges(int[] nums) {


      List<string> result = new List<string>();

      int n = nums.Length;
      if( n == 0 ){
        return result;
      }
      int start = nums[0];

      //nums[i] - nums[i- 1] = 1
      for ( int i = 1; i <= n; i++){
        if(i == n || nums[i] != nums[i -1] +1)
        {
          if ( start == nums[ i - 1]){
            result.Add(start.ToString());
            //7
          }
          else{
            result.Add(start + "->" + nums[i-1]);
             // 0 -> 2
          }
          if ( i < n){
            start = nums[i];
          }


        }
      }
      return result;
        
    }
}

File: 0228-summary-ranges/0228-summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
      if not nums:
        return []

      start = nums[0] # 0
      result = []

      for i in range(1, len(nums)):
        if nums[i] - nums[i - 1] != 1:
          if start == nums[i - 1]: #ie) [0,2,4]-> [0,2,4]
          #if its not consecutive, ie) 4-2 = 2 != 1, 
            result.append(str(start)) # append 7 will not happen, 
          #its because we can not add start since it might cont
          else:
            result.append(f"{start}->{nums[i-1]}")
          #0->2, 4->5
          start = nums[i] # start = 4
      # if start != nums[-1]:
      #   #since last index cannot be reach by index
      if start == nums[-1]:
        result.append(str(start))       # 혼자면 "7"
      else:
        result.append(f"{start}->{nums[-1]}")  # 연속이면 "7->9"
      return result







      